source.go

```diff
@@ -31,40 +32,50 @@ type Source struct {
 	sdk.UnimplementedSource
 
 	created int64
-	Config  Config
+	payload []byte
+	config  Config
 }
 
 func NewSource() sdk.Source {
 	return &Source{}
 }
 
-func (s *Source) Configure(ctx context.Context, config map[string]string) error {
+func (s *Source) Configure(_ context.Context, config map[string]string) error {
 	parsedCfg, err := Parse(config)
 	if err != nil {
 		return fmt.Errorf("invalid config: %w", err)
 	}
-	s.Config = parsedCfg
+	s.config = parsedCfg
 	return nil
 }
 
 func (s *Source) Open(ctx context.Context, position sdk.Position) error {
-	return nil // nothing to start
+	if s.config.PayloadFile == "" {
+		return nil // nothing to start
+	}
+
+	bytes, err := ioutil.ReadFile(s.config.PayloadFile)
```

Comment:
You are right there, it would need to be a file on the local file system. Currently, my thoughts on this are:
1. Uploading the file contents as part of the request definitely has its pros and solves the problem of remote files! 
2. Whatever HTTP server we use internally likely has a limit on the request size. On the other hand, we'd like to be able to use quite huge payloads. If there's an easy way to configure it for testing purposes, that'd be good.
3. More importantly, using a path makes it easier to create the test pipelines. Currently, we create them through a bash script. Loading a file into a variable would be possible. In future, though, I believe it would be good to move to pipeline config files. Setting those up with files "embedded" would be slightly more complex. 
4. We use the generator for testing purposes only anyway, so using the generator to access a file on a remote machine is probably a pretty rare use case. Mounting a file to a Docker container should

Reply:
> Mounting a file to a Docker container should be easy thing.

I'm not saying it's not easy, it's just not that elegant. I agree with your other points, getting the payload through the API would make our lives harder when writing benchmarks and tests, which is the main point of this connector. So let's do it this way for now.

We need to figure out how to handle these things generally though, letting the connector parse a local file is not ideal in other connectors - say we want to provide a certificate to the Kafka connector so it can use TLS. That's a file that needs to be provided through the API (granted, it's not that big, but for the sake of the discussion let's assume it can be arbitrarily big). Right now we pass it as a normal config option, but that can't be arbitrarily big. Also the point with the pipeline config file is valid, it wouldn't be possible to inject a file (you could do it with ENV variables, but that also has limits and is not a nice solution). So yeah, something we have yet to figure out I guess!

Reply:
> I'm not saying it's not easy, it's just not that elegant.

Definitely not that elegant! The reason why I said it's easy, is simply to compare the efforts vs. the values.



> We need to figure out how to handle these things generally though, letting the connector parse a local file is not ideal in other connectors

I agree :100: !